corehq/ex-submodules/casexml/apps/case/signals.py

```diff
@@ -13,5 +13,16 @@ def rebuild_form_cases(sender, xform, *args, **kwargs):
         rebuild_case_from_forms(domain, case_id, detail)
 
 
-xform_archived.connect(rebuild_form_cases)
-xform_unarchived.connect(rebuild_form_cases)
+connected = False
+
+
+def connect_signals():
+    global connected
+    if connected:
+        return
+    xform_archived.connect(rebuild_form_cases)
+    xform_unarchived.connect(rebuild_form_cases)
+    connected = True
```

Comment:
I get why you needed to keep the import (with `# noqa`), but can you fill me in on the need for this logic?

Reply:
I'm not familiar with the logic behind `connect`, but I felt there was a possibility, with multiple imports, to trigger the connect logic twice. If `connect` has no safeguards, then we'd end up triggering receivers twice. So, I felt it was safest to ensure this was only triggered once, no matter how many times the code was executed.